### ISSUE TITLE
PulleyPrimaryContentControllerDelegate not getting called

### ISSUE THREAD

**Issue by KingOlami** [52inc/Pulley#316]
I extended the pulley delegate and printed a value for everytime the BottomSheet changes from either collapsed to open but the delegate method does not work. I do not know why?

```
extension HomeVC: PulleyPrimaryContentControllerDelegate {
    func drawerPositionDidChange(drawer: PulleyViewController, bottomSafeArea: CGFloat) {
        
    }
    
    func makeUIAdjustmentsForFullscreen(progress: CGFloat, bottomSafeArea: CGFloat) {
        
    }
    
    func drawerChangedDistanceFromBottom(drawer: PulleyViewController, distance: CGFloat,
                                         bottomSafeArea: CGFloat) {
        
        print("DRAWER DISPLAY")
    }
}
```

**Comment by brendan09** [comment]
Have you implemented this in the primary content view controller or drawer view controller?

If you’re implementing this elsewhere you’ll need to assign that object to the delegate property of Pulley.

**Comment by KingOlami** [comment]
How please?

**Comment by brendan09** [comment]
PulleyViewController has a delegate property. Assign the object you want to be the delegate to that property.

**Comment by KingOlami** [comment]
Yes, that’s why I extended the viewcontroller as shown in my question

**Comment by brendan09** [comment]
Adding the delegate extension to a view controller doesn’t make it automatically receive delegate events. It must also be assigned to the delegate property.

If your view controller is your primary content view controller or drawer view controller you’ll need to ensure it uses the appropriate delegate for that view controller (primary vs drawer), it won’t work if you don’t implement the correct one. 

If your view controller is your primary content view controller or drawer view controller and it’s contained in a navigation controller (or tab bar controller) you’ll need to proxy the delegate methods through that controller.

**Comment by KingOlami** [comment]
I am implementing the correct one. I am lost and confused.

![screen shot 2019-02-26 at 10 37 46 am](https://user-images.githubusercontent.com/22158320/53407419-79b43980-39bc-11e9-983e-7be2b334f1f1.png)
![screen shot 2019-02-26 at 10 40 01 am](https://user-images.githubusercontent.com/22158320/53407446-8769bf00-39bc-11e9-890e-ce42dff598e4.png)

![screen shot 2019-02-26 at 10 38 38 am](https://user-images.githubusercontent.com/22158320/53407426-7e78ed80-39bc-11e9-817f-cdeace89a59e.png)


Any thing I am not implementing?

**Comment by KingOlami** [comment]
I have discovered the problem but solving it is what I do not know currently.so because I am using SWRevealController to show a side navigation menu, I do not know how to make the home controller the main controller

**Comment by brendan09** [comment]
First, I don’t support subclasses of Pulley as it can introduce issues. Please use PulleyViewController as-is.

It looks like your issue is that you’ve made the 2 things try to contain each other.

I’m not aware of how the slide to reveal library you’re using works, however you generally want to set it up like this:

Your window’s root view controller (or modally presented controller) should be the swipe to reveal controller, the swipe to reveal controller should contain PulleyViewController as one if it’s view controllers. Pulley must be initialized with a primary and drawer view controller that ARE NOT the swipe to reveal controller (or anything it manages). 

Think of it as building in layers. You can’t use a layer twice. Right now you’re using a layer twice: You’re having pulley try to contain the swipe to reveal controller AND having the swipe to reveal controller contain Pulley. That won’t work. 

If you don’t want Pulley to be IN the swipe to reveal controller, and instead want the swipe to reveal controller to live inside Pulley:

Pulley should be the root view controller of the window (or modally presented) and the primary content view controller should be the swipe to reveal view controller. The swipe to reveal controller can then contain a bottom / front VC as needed, neither of which is Pulley. IMPORTANT: In this case, the swipe to reveal controller IS the primary view controller, so that’s where the delegate events will be delivered. You’ll have to implement the delegate on that class. Most people choose to forward the delegate events on to the front controller in their implementation.

**Comment by brendan09** [comment]
There’s limited support I can provide for your use of 3rd party libraries. I can assist directly with Pulley, but am unable to generally assist with issues related to involving other libraries in the mix.
